src/main/java/com/minecolonies/client/gui/WindowTownHall.java

```diff
@@ -314,11 +473,84 @@ public void updateElement(int index, Pane rowPane)
                 rowPane.findPaneOfTypeByID("name", Label.class).setLabelText(citizen.getName());
             }
         });
+    }
+
+    /**
+     * Fills the workOrder list inside the townhall GUI.
+     */
+    private void fillWorkOrderList()
+    {
+        final ScrollingList workOrderList =  findPaneOfTypeByID(LIST_WORKORDER, ScrollingList.class);
+        workOrderList.enable();
+        workOrderList.show();
+
+        //Creates a dataProvider for the unemployed citizenList.
+        workOrderList.setDataProvider(new ScrollingList.DataProvider()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this anonymous class number of lines from 37 to at most 30, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1188)